js/autotrack/common.ts

```diff
@@ -1,19 +1,60 @@
 import * as _ from 'lodash';
 
+import { Component as ReactComponent } from 'react';
+import { Fiber as FiberNode } from 'react-reconciler';
 import { extractProps } from '../util/extractProps';
 import { BASE_HEAP_IGNORE_PROPS, getNextHeapIgnoreProps } from './heapIgnore';
 import { builtinPropExtractorConfig } from '../propExtractorConfig';
 
-// Returns an object containing a base set of component properties if we're not ignoring the
-// full interaction due to HeapIgnore.
+// The type definition of 'Component' from '@types/react' doesn't include the internal
+// '_reactInternalFiber' property, so create our own 'Component' type that includes this prop.
+// :TODO: (jmtaber129): Consider pulling this out if other TS files need this extended typing.
+interface Component extends ReactComponent {
+  _reactInternalFiber: FiberNode;
+}
+
+// Base properties for autotracked events.
+interface AutotrackProps {
+  touchableHierarchy: string;
+  targetText?: string;
+}
+
+interface HeapIgnoreProps {
+  ignoreInteraction?: boolean;
+  ignoreInnerHierarchy?: boolean;
+  ignoreAllProps?: boolean;
+  // :TODO: (jmtaber129): Add 'ignoreSpecificProps'.
+  ignoreTargetText?: boolean;
+}
+
+// The result of traversing up and back down a component's hierarchy.
+interface HierarchyResult {
+  hierarchy: string;
+  heapIgnoreProps: HeapIgnoreProps;
+}
+
+// The representation of a component in a Fiber hierarchy.
+interface ComponentHierarchyTraversalElement {
+  elementName: string;
+  fiberNode: FiberNode;
+  propsString: string;
+}
+
+// Returns an 'AutotrackProps' containing a base set of component properties if we're not ignoring
+// the full interaction due to HeapIgnore.
 // Returns null if we're ignoring the full interaction due to HeapIgnore.
-export const getBaseComponentProps = componentThis => {
+export const getBaseComponentProps: (
+  component: Component
+) => AutotrackProps | null = componentThis => {
   // Get the hierarchy traversal from root to target component, then get the actual hierarchy from
   // the traversal representation.
-  const touchableHierarchyTraversal = getComponentHierarchyTraversal(
+  const touchableHierarchyTraversal: ComponentHierarchyTraversalElement[] = getComponentHierarchyTraversal(
     componentThis
   );
-  const { hierarchy, heapIgnoreProps } = getHierarchyStringFromTraversal(
+  const {
+    hierarchy,
+    heapIgnoreProps,
+  }: HierarchyResult = getHierarchyStringFromTraversal(
```

Comment:
The explicit annotation here seems pretty unnecessary since you already know what values you're pulling out.

Reply:
fixed